# Introduction
If someone is looking to open a restaurant, where would they open it? What would they consider when selecting a location? In order to figure this out, I take New York City as an example and explore regional characteristics of restaurants. There are hundreds types of restaurants, based on the maximum total numbers among these restaurants, I choose Pizza Place for the following analysis. The analysis of other types of restaurants can be conducted with the same method.

In the following report, I mainly use Foursquare to acquire neighborhood information around restaurants in New York City. Besides the Foursquare location data, I obtain information of community boards from wikipedia:https://en.wikipedia.org/wiki/Neighborhoods_in_New_York_City.

With neighborhood information around restaurants, I cluster restaurants into different groups and explore if these groups has different performance according to customers reactions. And make suggestions to a restaurant owen to find a right place to start the business.

In [55]:
from bs4 import BeautifulSoup
import requests
import pandas as pd
from pandas.io.json import json_normalize

In [218]:
import lxml.html as lh
import numpy as np
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
from sklearn.cluster import KMeans
import matplotlib.cm as cm
import matplotlib.colors as colors

# Data
## Scrape boroughs from websites

In [13]:
url2='https://en.wikipedia.org/wiki/Neighborhoods_in_New_York_City'
source2=requests.get(url2).text
soup2=BeautifulSoup(source2,'lxml')

In [14]:
table_value=soup2.tbody
table_value=table_value.find_all('td')
table_value

[<td><a href="/wiki/Bronx_Community_Board_1" title="Bronx Community Board 1">Bronx CB 1</a></td>,
 <td align="right">7.17</td>,
 <td align="right">91,497</td>,
 <td align="right">12,761</td>,
 <td><a href="/wiki/Melrose,_Bronx" title="Melrose, Bronx">Melrose</a>, <a href="/wiki/Mott_Haven,_Bronx" title="Mott Haven, Bronx">Mott Haven</a>, <a href="/wiki/Port_Morris,_Bronx" title="Port Morris, Bronx">Port Morris</a>
 </td>,
 <td><a href="/wiki/Bronx_Community_Board_2" title="Bronx Community Board 2">Bronx CB 2</a></td>,
 <td align="right">5.54</td>,
 <td align="right">52,246</td>,
 <td align="right">9,792</td>,
 <td><a href="/wiki/Hunts_Point,_Bronx" title="Hunts Point, Bronx">Hunts Point</a>, <a href="/wiki/Longwood,_Bronx" title="Longwood, Bronx">Longwood</a>
 </td>,
 <td><a href="/wiki/Bronx_Community_Board_3" title="Bronx Community Board 3">Bronx CB 3</a></td>,
 <td align="right">4.07</td>,
 <td align="right">79,762</td>,
 <td align="right">19,598</td>,
 <td><a class="mw-redirect" hr

In [15]:
Cb=[]
Area=[]
Popc=[]
Popa=[]
Nei=[]
for n in range(0,len(table_value),5):
    
    Cb.append(table_value[n].text.replace('\xa0',' '))
    Area.append(table_value[n+1].text)
    Popc.append(table_value[n+2].text)
    Popa.append(table_value[n+3].text)
    Nei.append(table_value[n+4].text.split('\n')[0])

## Obtain Geological Location
Because geopy cannot recognize the address like 'Bronx CB 1', I use the first address in the list of Neighborhoods of each community board. If it is still not found, the second address will be used.

In [16]:
Lat=[]
Lon=[]
for i in range(0,len(Cb)):
    address ='{}, New York City, NY'.format(Nei[i].split(',')[0])
    geolocator = Nominatim(user_agent="ny_explorer")
    try:
        location = geolocator.geocode(address)
        Lat.append(location.latitude)
        Lon.append(location.longitude)
    except:
        address ='{}, New York City, NY'.format(Nei[i].split(',')[1])
        location = geolocator.geocode(address)
        Lat.append(location.latitude)
        Lon.append(location.longitude)

In [17]:
ar_f=[]
for cb,area,popc,popa,nei,lat,lon in zip(Cb,Area,Popc,Popa,Nei,Lat,Lon):
    ar_f.append({'Community Board':cb,'Area/km2':area,'Pop.Census':popc,'Pop/km2':popa,'Neighborhoods':nei,
                'Latitude':lat,'Longitude':lon})
df2= pd.DataFrame(ar_f)
df2=df2[['Community Board','Area/km2','Pop.Census','Pop/km2','Neighborhoods','Latitude','Longitude']]
df2

Community Board Area/km2 Pop.Census Pop/km2  \
0           Bronx CB 1     7.17     91,497  12,761   
1           Bronx CB 2     5.54     52,246   9,792   
2           Bronx CB 3     4.07     79,762  19,598   
3           Bronx CB 4     5.28    146,441  27,735   
4           Bronx CB 5     3.55    128,200  36,145   
5           Bronx CB 6     4.01     83,268  20,765   
6           Bronx CB 7     4.84    139,286  28,778   
7           Bronx CB 8     8.83    101,731  11,521   
8           Bronx CB 9    12.41    172,298  13,884   
9          Bronx CB 10    16.76    120,392   7,183   
10         Bronx CB 11     9.32    113,232  12,149   
11         Bronx CB 12    14.56    152,344  10,463   
12       Brooklyn CB 1    12.82    160,338  12,507   
13       Brooklyn CB 2     7.72     98,620  12,775   
14       Brooklyn CB 3     7.67    143,867  18,757   
15       Brooklyn CB 4     5.31    104,358  19,653   
16       Brooklyn CB 5    14.61    173,198  11,855   
17       Brooklyn CB 6     9.01    104,054  11,549   
18       Brooklyn CB 7    10.96    120,063  10,955   
19       Brooklyn CB 8     4.25     96,076  22,606   
20       Brooklyn CB 9     4.07    104,014  25,556   
21      Brooklyn CB 10    10.57    122,542  11,593   
22      Brooklyn CB 11    10.18    172,129  16,909   
23      Brooklyn CB 12     9.32    185,046  19,855   
24      Brooklyn CB 13     8.88    106,120  11,950   
25      Brooklyn CB 14     7.61    168,806  22,182   
26      Brooklyn CB 15    12.82    160,319  12,505   
27      Brooklyn CB 16     4.97     85,343  17,172   
28      Brooklyn CB 17     8.73    165,753  18,987   
29      Brooklyn CB 18    24.68    194,653   7,887   
30      Manhattan CB 1     4.45     34,420   7,735   
31      Manhattan CB 2     4.01     93,119  23,222   
32      Manhattan CB 3     4.56    164,407  36,054   
33      Manhattan CB 4     5.41     87,479  16,170   
34      Manhattan CB 5     4.25     44,028  10,360   
35      Manhattan CB 6     3.55    136,152  38,353   
36      Manhattan CB 7     5.46    207,699  38,040   
37      Manhattan CB 8     5.13    217,063  42,312   
38      Manhattan CB 9     3.91    111,724  28,574   
39     Manhattan CB 10     3.63    107,109  29,507   
40     Manhattan CB 11     5.75    117,743  20,477   
41     Manhattan CB 12     7.64    208,414  27,279   
42         Queens CB 1    14.97    211,220  14,110   
43         Queens CB 2    13.36    109,920   8,228   
44         Queens CB 3     7.33    169,083  23,067   
45         Queens CB 4     6.16    167,005  27,111   
46         Queens CB 5    19.63    165,911   8,452   
47         Queens CB 6     7.59    115,967  15,279   
48         Queens CB 7    32.97    242,952   7,369   
49         Queens CB 8    19.17    146,594   7,647   
50         Queens CB 9    10.08    141,608  14,048   
51        Queens CB 10    16.19    127,274   7,861   
52        Queens CB 11    24.99    116,404   4,658   
53        Queens CB 12    25.46    223,602   8,782   
54        Queens CB 13    33.31    196,284   5,893   
55        Queens CB 14    23.28    106,686   4,583   
56  Staten Island CB 1    36.62    162,609   4,440   
57  Staten Island CB 2    54.62    127,071   2,326   
58  Staten Island CB 3    58.97    152,908   2,593   
59       New York City   831.39  8,008,278   9,632   

                                        Neighborhoods   Latitude  Longitude  
0                    Melrose, Mott Haven, Port Morris  40.824545 -73.910414  
1                               Hunts Point, Longwood  40.812601 -73.884025  
2   Claremont, Concourse Village, Crotona Park, Mo...  40.839876 -73.907328  
3                               Concourse, Highbridge  40.874217 -73.890410  
4   Fordham, Morris Heights, Mount Hope, Universit...  40.859267 -73.898469  
5         Bathgate, Belmont, East Tremont, West Farms  40.855278 -73.886389  
6           Bedford Park, Norwood, University Heights  40.870100 -73.885691  
7   Fieldston, Kingsbridge, Kingsbridge Heights, M...  40.894444 -73.903333  
8   Bronx 

In [18]:
df2.drop(index=59,inplace=True)#delete the last row which represents the whole city
df2

Community Board Area/km2 Pop.Census Pop/km2  \
0           Bronx CB 1     7.17     91,497  12,761   
1           Bronx CB 2     5.54     52,246   9,792   
2           Bronx CB 3     4.07     79,762  19,598   
3           Bronx CB 4     5.28    146,441  27,735   
4           Bronx CB 5     3.55    128,200  36,145   
5           Bronx CB 6     4.01     83,268  20,765   
6           Bronx CB 7     4.84    139,286  28,778   
7           Bronx CB 8     8.83    101,731  11,521   
8           Bronx CB 9    12.41    172,298  13,884   
9          Bronx CB 10    16.76    120,392   7,183   
10         Bronx CB 11     9.32    113,232  12,149   
11         Bronx CB 12    14.56    152,344  10,463   
12       Brooklyn CB 1    12.82    160,338  12,507   
13       Brooklyn CB 2     7.72     98,620  12,775   
14       Brooklyn CB 3     7.67    143,867  18,757   
15       Brooklyn CB 4     5.31    104,358  19,653   
16       Brooklyn CB 5    14.61    173,198  11,855   
17       Brooklyn CB 6     9.01    104,054  11,549   
18       Brooklyn CB 7    10.96    120,063  10,955   
19       Brooklyn CB 8     4.25     96,076  22,606   
20       Brooklyn CB 9     4.07    104,014  25,556   
21      Brooklyn CB 10    10.57    122,542  11,593   
22      Brooklyn CB 11    10.18    172,129  16,909   
23      Brooklyn CB 12     9.32    185,046  19,855   
24      Brooklyn CB 13     8.88    106,120  11,950   
25      Brooklyn CB 14     7.61    168,806  22,182   
26      Brooklyn CB 15    12.82    160,319  12,505   
27      Brooklyn CB 16     4.97     85,343  17,172   
28      Brooklyn CB 17     8.73    165,753  18,987   
29      Brooklyn CB 18    24.68    194,653   7,887   
30      Manhattan CB 1     4.45     34,420   7,735   
31      Manhattan CB 2     4.01     93,119  23,222   
32      Manhattan CB 3     4.56    164,407  36,054   
33      Manhattan CB 4     5.41     87,479  16,170   
34      Manhattan CB 5     4.25     44,028  10,360   
35      Manhattan CB 6     3.55    136,152  38,353   
36      Manhattan CB 7     5.46    207,699  38,040   
37      Manhattan CB 8     5.13    217,063  42,312   
38      Manhattan CB 9     3.91    111,724  28,574   
39     Manhattan CB 10     3.63    107,109  29,507   
40     Manhattan CB 11     5.75    117,743  20,477   
41     Manhattan CB 12     7.64    208,414  27,279   
42         Queens CB 1    14.97    211,220  14,110   
43         Queens CB 2    13.36    109,920   8,228   
44         Queens CB 3     7.33    169,083  23,067   
45         Queens CB 4     6.16    167,005  27,111   
46         Queens CB 5    19.63    165,911   8,452   
47         Queens CB 6     7.59    115,967  15,279   
48         Queens CB 7    32.97    242,952   7,369   
49         Queens CB 8    19.17    146,594   7,647   
50         Queens CB 9    10.08    141,608  14,048   
51        Queens CB 10    16.19    127,274   7,861   
52        Queens CB 11    24.99    116,404   4,658   
53        Queens CB 12    25.46    223,602   8,782   
54        Queens CB 13    33.31    196,284   5,893   
55        Queens CB 14    23.28    106,686   4,583   
56  Staten Island CB 1    36.62    162,609   4,440   
57  Staten Island CB 2    54.62    127,071   2,326   
58  Staten Island CB 3    58.97    152,908   2,593   

                                        Neighborhoods   Latitude  Longitude  
0                    Melrose, Mott Haven, Port Morris  40.824545 -73.910414  
1                               Hunts Point, Longwood  40.812601 -73.884025  
2   Claremont, Concourse Village, Crotona Park, Mo...  40.839876 -73.907328  
3                               Concourse, Highbridge  40.874217 -73.890410  
4   Fordham, Morris Heights, Mount Hope, Universit...  40.859267 -73.898469  
5         Bathgate, Belmont, East Tremont, West Farms  40.855278 -73.886389  
6           Bedford Park, Norwood, University Heights  40.870100 -73.885691  
7   Fieldston, Kingsbridge, Kingsbridge Heights, M...  40.894444 -73.903333  
8   Bronx River, Bruckner, Castle Hill, Clason Poi...  40.850485

## Obtain Venues in Neighborhoods 

In [19]:
CLIENT_ID = 'QPG4LS2XMXGC2KXOGFMHKXCTIACEGEYEKGT0BS3E0DAOXGDG' # your Foursquare ID
CLIENT_SECRET = 'WVWNOOZAJ11DOKZAKJWKPPD1LRRESFJFZ4BABL5RBDD50UT1' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: QPG4LS2XMXGC2KXOGFMHKXCTIACEGEYEKGT0BS3E0DAOXGDG
CLIENT_SECRET:WVWNOOZAJ11DOKZAKJWKPPD1LRRESFJFZ4BABL5RBDD50UT1


In [52]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    LIMIT = 100
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['id'],
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude',
                  'id',
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [53]:
map_result=getNearbyVenues(names=df2['Community Board'],
                           latitudes=df2['Latitude'],
                          longitudes=df2['Longitude'])

Bronx CB 1
Bronx CB 2
Bronx CB 3
Bronx CB 4
Bronx CB 5
Bronx CB 6
Bronx CB 7
Bronx CB 8
Bronx CB 9
Bronx CB 10
Bronx CB 11
Bronx CB 12
Brooklyn CB 1
Brooklyn CB 2
Brooklyn CB 3
Brooklyn CB 4
Brooklyn CB 5
Brooklyn CB 6
Brooklyn CB 7
Brooklyn CB 8
Brooklyn CB 9
Brooklyn CB 10
Brooklyn CB 11
Brooklyn CB 12
Brooklyn CB 13
Brooklyn CB 14
Brooklyn CB 15
Brooklyn CB 16
Brooklyn CB 17
Brooklyn CB 18
Manhattan CB 1
Manhattan CB 2
Manhattan CB 3
Manhattan CB 4
Manhattan CB 5
Manhattan CB 6
Manhattan CB 7
Manhattan CB 8
Manhattan CB 9
Manhattan CB 10
Manhattan CB 11
Manhattan CB 12
Queens CB 1
Queens CB 2
Queens CB 3
Queens CB 4
Queens CB 5
Queens CB 6
Queens CB 7
Queens CB 8
Queens CB 9
Queens CB 10
Queens CB 11
Queens CB 12
Queens CB 13
Queens CB 14
Staten Island CB 1
Staten Island CB 2
Staten Island CB 3


In [54]:
map_result

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0             Bronx CB 1              40.824545              -73.910414   
1             Bronx CB 1              40.824545              -73.910414   
2             Bronx CB 1              40.824545              -73.910414   
3             Bronx CB 1              40.824545              -73.910414   
4             Bronx CB 1              40.824545              -73.910414   
5             Bronx CB 1              40.824545              -73.910414   
6             Bronx CB 1              40.824545              -73.910414   
7             Bronx CB 1              40.824545              -73.910414   
8             Bronx CB 1              40.824545              -73.910414   
9             Bronx CB 1              40.824545              -73.910414   
10            Bronx CB 1              40.824545              -73.910414   
11            Bronx CB 1              40.824545              -73.910414   
12            Bronx CB 1              40.824545              -73.910414   
13            Bronx CB 1              40.824545              -73.910414   
14            Bronx CB 1              40.824545              -73.910414   
15            Bronx CB 1              40.824545              -73.910414   
16            Bronx CB 1              40.824545              -73.910414   
17            Bronx CB 1              40.824545              -73.910414   
18            Bronx CB 1              40.824545              -73.910414   
19            Bronx CB 1              40.824545              -73.910414   
20            Bronx CB 1              40.824545              -73.910414   
21            Bronx CB 1              40.824545              -73.910414   
22            Bronx CB 1              40.824545              -73.910414   
23            Bronx CB 2              40.812601              -73.884025   
24            Bronx CB 2              40.812601              -73.884025   
25            Bronx CB 2              40.812601              -73.884025   
26            Bronx CB 2              40.812601              -73.884025   
27            Bronx CB 2              40.812601              -73.884025   
28            Bronx CB 2              40.812601              -73.884025   
29            Bronx CB 2              40.812601              -73.884025   
...                  ...                    ...                     ...   
2527  Staten Island CB 1              40.632326              -74.165144   
2528  Staten Island CB 2              40.598438              -74.072641   
2529  Staten Island CB 2              40.598438              -74.072641   
2530  Staten Island CB 2              40.598438              -74.072641   
2531  Staten Island CB 2              40.598438              -74.072641   
2532  Staten Island CB 2              40.598438              -74.072641   
2533  Staten Island CB 2              40.598438              -74.072641   
2534  Staten Island CB 2              40.598438              -74.072641   
2535  Staten Island CB 2              40.598438              -74.072641   
2536  Staten Island CB 3              40.544550              -74.176532   
2537  Staten Island CB 3              40.544550              -74.176532   
2538  Staten Island CB 3              40.544550              -74.176532   
2539  Staten Island CB 3              40.544550              -74.176532   
2540  Staten Island CB 3              40.544550              -74.176532   
2541  Staten Island CB 3              40.544550              -74.176532   
2542  Staten Island CB 3              40.544550              -74.176532   
2543  Staten Island CB 3              40.544550              -74.176532   
2544  Staten Island CB 3              40.544550              -74.176532   
2545  Staten Island CB 3              40.544550              -74.176532   
2546  Staten Island CB 3              40.544550              -74.176532   
2547  Staten Island CB 3              40.544550              -74.176532   
2548  Staten Island CB 3             

In [23]:
map_result.shape

(2555, 7)

In [25]:
map_result['Venue Category'].unique()

array(['Italian Restaurant', 'Diner', 'Mexican Restaurant', 'Art Gallery',
       'Fried Chicken Joint', 'Fast Food Restaurant', 'Supermarket',
       'Donut Shop', 'Pharmacy', 'Hotel', 'Bus Station',
       'Check Cashing Service', 'Pizza Place', 'Laundromat',
       'Chinese Restaurant', 'Food Truck', 'Juice Bar', 'Deli / Bodega',
       'Discount Store', 'School', 'Spanish Restaurant', 'Bank',
       'Farmers Market', 'BBQ Joint', 'Restaurant', 'Waste Facility',
       'Park', 'Grocery Store', 'Moving Target', 'Pub', 'Theater', 'Gym',
       'Sandwich Place', 'Metro Station', 'Latin American Restaurant',
       'Gym / Fitness Center', 'African Restaurant', 'Music Venue',
       'Mobile Phone Shop', 'Bakery', 'Ice Cream Shop', 'Clothing Store',
       'Video Game Store', 'Sporting Goods Shop', 'Shoe Store',
       'Supplement Shop', 'Kids Store', 'Greek Restaurant',
       'Accessories Store', 'Electronics Store', 'Liquor Store',
       'Department Store', 'Thrift / Vintage Store', '

In [80]:
rr_df=map_result[map_result['Venue Category'].str.contains("Restaurant")]
rr_df

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0             Bronx CB 1              40.824545              -73.910414   
2             Bronx CB 1              40.824545              -73.910414   
5             Bronx CB 1              40.824545              -73.910414   
12            Bronx CB 1              40.824545              -73.910414   
24            Bronx CB 2              40.812601              -73.884025   
30            Bronx CB 2              40.812601              -73.884025   
36            Bronx CB 3              40.839876              -73.907328   
42            Bronx CB 4              40.874217              -73.890410   
46            Bronx CB 4              40.874217              -73.890410   
50            Bronx CB 4              40.874217              -73.890410   
60            Bronx CB 4              40.874217              -73.890410   
65            Bronx CB 5              40.859267              -73.898469   
67            Bronx CB 5              40.859267              -73.898469   
75            Bronx CB 5              40.859267              -73.898469   
80            Bronx CB 5              40.859267              -73.898469   
83            Bronx CB 5              40.859267              -73.898469   
85            Bronx CB 5              40.859267              -73.898469   
90            Bronx CB 5              40.859267              -73.898469   
106           Bronx CB 5              40.859267              -73.898469   
108           Bronx CB 5              40.859267              -73.898469   
111           Bronx CB 5              40.859267              -73.898469   
116           Bronx CB 5              40.859267              -73.898469   
119           Bronx CB 5              40.859267              -73.898469   
122           Bronx CB 5              40.859267              -73.898469   
135           Bronx CB 6              40.855278              -73.886389   
139           Bronx CB 6              40.855278              -73.886389   
148           Bronx CB 6              40.855278              -73.886389   
152           Bronx CB 6              40.855278              -73.886389   
154           Bronx CB 6              40.855278              -73.886389   
155           Bronx CB 6              40.855278              -73.886389   
...                  ...                    ...                     ...   
2414        Queens CB 10              40.657882              -73.836246   
2415        Queens CB 10              40.657882              -73.836246   
2416        Queens CB 10              40.657882              -73.836246   
2418        Queens CB 10              40.657882              -73.836246   
2419        Queens CB 10              40.657882              -73.836246   
2423        Queens CB 10              40.657882              -73.836246   
2424        Queens CB 10              40.657882              -73.836246   
2430        Queens CB 10              40.657882              -73.836246   
2433        Queens CB 10              40.657882              -73.836246   
2434        Queens CB 10              40.657882              -73.836246   
2435        Queens CB 10              40.657882              -73.836246   
2442        Queens CB 11              40.761452              -73.789724   
2446        Queens CB 11              40.761452              -73.789724   
2447        Queens CB 11              40.761452              -73.789724   
2452        Queens CB 11              40.761452              -73.789724   
2453        Queens CB 11              40.761452              -73.789724   
2454        Queens CB 11              40.761452              -73.789724   
2466        Queens CB 11              40.761452              -73.789724   
2469        Queens CB 12              40.691485              -73.805677   
2478        Queens CB 12              40.691485              -73.805677   
2487        Queens CB 12              40.691485              -73.805677   
2494        Queens CB 13             

Select Pizza Place as the analysis target. There are 117 pizza places in New York City, which is the most among various types of restaurants.

In [44]:
cat_freq=pd.value_counts(map_result['Venue Category'])
cat_freq.head(80)

Pizza Place                        117
Italian Restaurant                  81
Bakery                              73
Deli / Bodega                       70
Coffee Shop                         66
Chinese Restaurant                  58
Café                                52
Mexican Restaurant                  48
Bar                                 44
Sandwich Place                      43
Pharmacy                            39
American Restaurant                 39
Gym                                 37
Art Gallery                         36
Park                                36
Grocery Store                       35
Donut Shop                          33
Bank                                33
Cocktail Bar                        31
Ice Cream Shop                      30
Sushi Restaurant                    30
Wine Shop                           26
Hotel                               26
Gym / Fitness Center                25
Mobile Phone Shop                   25
Latin American Restaurant

In [84]:
pp_df=map_result[map_result['Venue Category'].str.contains("Pizza")]
pp_df

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
11            Bronx CB 1              40.824545              -73.910414   
19            Bronx CB 1              40.824545              -73.910414   
27            Bronx CB 2              40.812601              -73.884025   
40            Bronx CB 4              40.874217              -73.890410   
72            Bronx CB 5              40.859267              -73.898469   
113           Bronx CB 5              40.859267              -73.898469   
123           Bronx CB 5              40.859267              -73.898469   
127           Bronx CB 5              40.859267              -73.898469   
134           Bronx CB 5              40.859267              -73.898469   
138           Bronx CB 6              40.855278              -73.886389   
143           Bronx CB 6              40.855278              -73.886389   
161           Bronx CB 6              40.855278              -73.886389   
177           Bronx CB 6              40.855278              -73.886389   
185           Bronx CB 6              40.855278              -73.886389   
190           Bronx CB 6              40.855278              -73.886389   
200           Bronx CB 7              40.870100              -73.885691   
205           Bronx CB 7              40.870100              -73.885691   
221           Bronx CB 7              40.870100              -73.885691   
222           Bronx CB 7              40.870100              -73.885691   
254           Bronx CB 9              40.850485              -73.840404   
269          Bronx CB 10              40.847456              -73.786466   
298          Bronx CB 12              40.861222              -73.841286   
332        Brooklyn CB 1              40.723713              -73.950971   
370        Brooklyn CB 1              40.723713              -73.950971   
392        Brooklyn CB 1              40.723713              -73.950971   
498        Brooklyn CB 3              40.683436              -73.941249   
499        Brooklyn CB 3              40.683436              -73.941249   
504        Brooklyn CB 3              40.683436              -73.941249   
524        Brooklyn CB 3              40.683436              -73.941249   
530        Brooklyn CB 3              40.683436              -73.941249   
...                  ...                    ...                     ...   
1925      Manhattan CB 9              40.824145              -73.950062   
1939      Manhattan CB 9              40.824145              -73.950062   
1944      Manhattan CB 9              40.824145              -73.950062   
1948      Manhattan CB 9              40.824145              -73.950062   
1953     Manhattan CB 10              40.807879              -73.945415   
1960     Manhattan CB 10              40.807879              -73.945415   
2030     Manhattan CB 10              40.807879              -73.945415   
2068     Manhattan CB 11              40.794722              -73.942500   
2073     Manhattan CB 11              40.794722              -73.942500   
2074     Manhattan CB 11              40.794722              -73.942500   
2090     Manhattan CB 11              40.794722              -73.942500   
2124     Manhattan CB 12              40.869258              -73.920495   
2142         Queens CB 1              40.772014              -73.930267   
2148         Queens CB 1              40.772014              -73.930267   
2157         Queens CB 1              40.772014              -73.930267   
2195         Queens CB 4              40.746959              -73.860146   
2198         Queens CB 4              40.746959              -73.860146   
2208         Queens CB 4              40.746959              -73.860146   
2264         Queens CB 5              40.707047              -73.903470   
2305         Queens CB 6              40.719577              -73.844869   
2308         Queens CB 6              40.719577              -73.844869   
2372         Queens CB 9             

By venue_id, using Foursquare to acquire detailed information of each pizza place.

In [154]:
venue_id='4aabd3e6f964a5204a5a20e3'
url_venue = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
result = requests.get(url_venue).json()
result

{'meta': {'code': 200, 'requestId': '5db5c889342adf0038fad354'},
 'response': {'venue': {'id': '4aabd3e6f964a5204a5a20e3',
   'name': 'Zero Otto Nove',
   'contact': {'phone': '7182201027',
    'formattedPhone': '(718) 220-1027',
    'facebook': '129459203781025',
    'facebookUsername': 'ZeroOttoNove',
    'facebookName': 'Zero Otto Nove'},
   'location': {'address': '2357 Arthur Ave',
    'crossStreet': 'at E 186th St',
    'lat': 40.85471353015844,
    'lng': -73.88838751470229,
    'labeledLatLngs': [{'label': 'display',
      'lat': 40.85471353015844,
      'lng': -73.88838751470229}],
    'postalCode': '10458',
    'cc': 'US',
    'city': 'Bronx',
    'state': 'NY',
    'country': 'United States',
    'formattedAddress': ['2357 Arthur Ave (at E 186th St)',
     'Bronx, NY 10458',
     'United States']},
   'canonicalUrl': 'https://foursquare.com/v/zero-otto-nove/4aabd3e6f964a5204a5a20e3',
   'categories': [{'id': '4bf58dd8d48988d1ca941735',
     'name': 'Pizza Place',
     'plura

Select out Rating, Price, Likes, Photos, Tips into a dataframe. And drop those places without a rating.

In [114]:
pp=pp_df['id']
venue_info_lst=[]
for venue_id in pp:
    url_v = 'https://api.foursquare.com/v2/venues/{}?client_id={}&client_secret={}&v={}'.format(venue_id, CLIENT_ID, CLIENT_SECRET, VERSION)
    result = requests.get(url_v).json()
    name=result['response']['venue']['name']
    try:
        rating=result['response']['venue']['rating']
    except:
        rating=0
    try:
        price=result['response']['venue']['price']['tier']
    except:
        price=0
    try:
        like=result['response']['venue']['likes']['count']
    except:
        like=0
    try:
        photo=result['response']['venue']['photos']['count']
    except:
        photo=0
    try:
        tips=result['response']['venue']['tips']['count']
    except:
        tips=0
    
    venue_info_lst.append([venue_id,name,rating, price, like, photo, tips])
venue_info_df=pd.DataFrame([venue_info for venue_info in venue_info_lst])
venue_info_df.columns = ['id', 'Name', 'Rating','Price', 'Likes', 'Photos', 'Tips']
venue_info_df


id                                      Name  \
0    4e4cf9c1bd413c4cc66dae95                          Linda's Pizzeria   
1    4df7be37aeb7f7c3b5436f7a                      Little Caesars Pizza   
2    4ca3e96c5720b1f78d0936ef                      Fratellis Pizza Cafe   
3    4afac66af964a520a11822e3                            Jerome's Pizza   
4    4bf329ef2d62952165ec5f58                        Best Italian Pizza   
5    4cb0f72d75ebb60cba60c9ad                            Domino's Pizza   
6    4ca680d1b7106dcba0d35ea5                             Susie's Pizza   
7    513fd3c3e4b0a8eca5f8b719                    Joey Pepperoni's Pizza   
8    5a272952fe37404aed37cf4e                               Bella Pizza   
9    4aabd3e6f964a5204a5a20e3                            Zero Otto Nove   
10   4aee1220f964a520ced121e3                        Full Moon Pizzeria   
11   4b140a90f964a520489c23e3                           Bellini's Pizza   
12   4b2a6ba8f964a520d9a824e3                             Pugsley Pizza   
13   4d03ef588620224bff0ba640            Catania's Pizzeria & Cafe Inc.   
14   575adfe8cd10afcbbf8d1a66                               Bella Pizza   
15   4bd71c916798ef3b563a668d                     My Place Family Pizza   
16   4afac66af964a520a11822e3                            Jerome's Pizza   
17   53ab09df498e91a454e6d3a5                             Bedford Pizza   
18   4c5db38e6ebe2d7faca1d32e                  Rocco's Pizza Restaurant   
19   5169feaae4b04280a40bcd5a                 The Famous Emilio's Pizza   
20   4c27a574c11dc9b634f32a24                  Filomena's Pizza & Pasta   
21   5054e648e4b096ad99b120e1                          Carbo's Pizzeria   
22   5761f36b498e2f6dca191e23                       Screamer's Pizzeria   
23   4ab947baf964a520c17e20e3                                Russ Pizza   
24   4a142410f964a52007781fe3                               Italy Pizza   
25   4a593de0f964a52015b91fe3                                 Saraghina   
26   58bad1af45c3ed4ca99f97f9                             Bravi Ragazzi   
27   4ecc3ea22c5b26aab86608f8                           Rocco Pizza III   
28   4b88126af964a52096dd31e3                            Domino's Pizza   
29   4e3f52476365a251c704391c                      Little Caesars Pizza   
..                        ...                                       ...   
87   4d5028cedcce224bb140d51b                              Olga's Pizza   
88   5812ea4538fa932fe0cc3951                         Papa John's Pizza   
89   505380aee4b07ab78c6545fe                     Vittorio's Pizza Plus   
90   4df94db2a809fe7742217cf6                                 P&M Pizza   
91   55a4456a498e29417f5c71a2                                babbalucci   
92   56e86a57cd1017cb53f3e8f9               Sottocasa Pizzeria - Harlem   
93   4ec312c6be7b4ee6107f53be                         Papa John's Pizza   
94   4bffe7dd92a6c928d44443e2                        La Nostra Pizzeria   
95   4ac54941f964a5205cb120e3                     Sam's Famous Pizzeria   
96   4bae85aff964a5206bbc3be3                         Milano's Pizzeria   
97   4c0ab529a1b32d7fd8d599f0                            Domino's Pizza   
98   4ab97430f964a520727f20e3                Grandpa's Brick Oven Pizza   
99   4bf2ad7782002d7fd5cfd0bb          CHUBBY BURGERS chicken and pizza   
100  50ece8c6e4b0ffbaa8a65f57                            Vidali's Pizza   
101  4e4e5b02bd4101d0d7a8626f                       S & S Calabro Pizza   
102  4bcb3f9ccc8cd13aa0f2becf              Tony's Pizzeria & Restaurant   
103  4b7c363df964a520ce842fe3               Angelo's Pizza & Restaurant   
104  4d3bd7f2fb4c5481ca6ccf14                         Papa John's Pizza   
105  5431ba1c498ec8da8a2e049e             Julie's Pizzeria & Restaurant   
106  4f6cb7a0e4b0a6b770c9b395                              Gloria Pizza   
107  45ac11b0f964a5205b411fe3                              Nick's Pizza   
108  4baebed0f964a5204ad43be3                       

In [184]:
pp_dff=venue_info_df[venue_info_df['Rating']!=0.0]
pp_dff

id                                      Name  \
1    4df7be37aeb7f7c3b5436f7a                      Little Caesars Pizza   
2    4ca3e96c5720b1f78d0936ef                      Fratellis Pizza Cafe   
3    4afac66af964a520a11822e3                            Jerome's Pizza   
4    4bf329ef2d62952165ec5f58                        Best Italian Pizza   
5    4cb0f72d75ebb60cba60c9ad                            Domino's Pizza   
9    4aabd3e6f964a5204a5a20e3                            Zero Otto Nove   
10   4aee1220f964a520ced121e3                        Full Moon Pizzeria   
11   4b140a90f964a520489c23e3                           Bellini's Pizza   
12   4b2a6ba8f964a520d9a824e3                             Pugsley Pizza   
13   4d03ef588620224bff0ba640            Catania's Pizzeria & Cafe Inc.   
15   4bd71c916798ef3b563a668d                     My Place Family Pizza   
16   4afac66af964a520a11822e3                            Jerome's Pizza   
19   5169feaae4b04280a40bcd5a                 The Famous Emilio's Pizza   
20   4c27a574c11dc9b634f32a24                  Filomena's Pizza & Pasta   
22   5761f36b498e2f6dca191e23                       Screamer's Pizzeria   
23   4ab947baf964a520c17e20e3                                Russ Pizza   
24   4a142410f964a52007781fe3                               Italy Pizza   
25   4a593de0f964a52015b91fe3                                 Saraghina   
26   58bad1af45c3ed4ca99f97f9                             Bravi Ragazzi   
27   4ecc3ea22c5b26aab86608f8                           Rocco Pizza III   
28   4b88126af964a52096dd31e3                            Domino's Pizza   
29   4e3f52476365a251c704391c                      Little Caesars Pizza   
30   4eb5e667b8030fb583171e69                      Little Caesars Pizza   
31   4c88221997828cfab0b399aa                         Papa John's Pizza   
32   5d56bcea6ac2400008de14d9                              F&F Pizzeria   
33   596ab0631de7657c78f4251b                     Brooklyn Pizza Market   
34   49d7e8c0f964a520795d1fe3                            Giardini Pizza   
35   58016254d67c1565bc122795                        Via Roma Pizza Bar   
36   4cd896b4da85224b80604dca                        Charles's Pizzeria   
37   4ca7cf4414c337048999cd3b                             Luigi's Pizza   
..                        ...                                       ...   
82   43c4bb46f964a5205a2d1fe3                                     Posto   
83   5893583c1470b050ba899197                          Donna Margherita   
84   4a72520af964a520e2da1fe3                                Pizza Park   
85   5b33c56f06fb600032c88e26                                Oath Pizza   
86   4c0fde4b81e976b0623c0eeb   La Crosta Restaurant & Gourmet Pizzeria   
87   4d5028cedcce224bb140d51b                              Olga's Pizza   
88   5812ea4538fa932fe0cc3951                         Papa John's Pizza   
89   505380aee4b07ab78c6545fe                     Vittorio's Pizza Plus   
90   4df94db2a809fe7742217cf6                                 P&M Pizza   
91   55a4456a498e29417f5c71a2                                babbalucci   
92   56e86a57cd1017cb53f3e8f9               Sottocasa Pizzeria - Harlem   
93   4ec312c6be7b4ee6107f53be                         Papa John's Pizza   
94   4bffe7dd92a6c928d44443e2                        La Nostra Pizzeria   
95   4ac54941f964a5205cb120e3                     Sam's Famous Pizzeria   
96   4bae85aff964a5206bbc3be3                         Milano's Pizzeria   
97   4c0ab529a1b32d7fd8d599f0                            Domino's Pizza   
98   4ab97430f964a520727f20e3                Grandpa's Brick Oven Pizza   
99   4bf2ad7782002d7fd5cfd0bb          CHUBBY BURGERS chicken and pizza   
100  50ece8c6e4b0ffbaa8a65f57                            Vidali's Pizza   
102  4bcb3f9ccc8cd13aa0f2becf              Tony's Pizzeria & Restaurant   
103  4b7c363df964a520ce842fe3               Angelo's Pizza & Restaurant   
104  4d3bd7f2fb4c5481ca6ccf14                       

In [185]:
pp_info_cor=pp_dff.drop(['id','Name'],1)
pp_info_cor.corr()

Rating     Price     Likes    Photos      Tips
Rating  1.000000  0.468022  0.493203  0.450074  0.510075
Price   0.468022  1.000000  0.624174  0.528872  0.713245
Likes   0.493203  0.624174  1.000000  0.868937  0.892418
Photos  0.450074  0.528872  0.868937  1.000000  0.829932
Tips    0.510075  0.713245  0.892418  0.829932  1.000000

Showing from the correlation matrix, Likes, Photos and Tips are highly correlated to each other. But Likes is not highly related to Rating. Customers who click Likes for some specific reasons but give lower ratings to the general performance might cause this low correlation. Therefore I choose Rating to represent the restaurant. Rating is somewhat correlated to Price, which indicates that the price might not affect impressions of customers on that place significantly.

# Obtain Neighborhood Info of Restaurants

In [159]:
pp_result=getNearbyVenues(names=pp_df['Venue'],
                           latitudes=pp_df['Venue Latitude'],
                          longitudes=pp_df['Venue Longitude'])

Linda's Pizzeria
Little Caesars Pizza
Fratellis Pizza Cafe
Jerome's Pizza
Best Italian Pizza
Domino's Pizza
Susie's Pizza
Joey Pepperoni's Pizza
Bella Pizza
Zero Otto Nove
Full Moon Pizzeria
Bellini's Pizza
Pugsley Pizza
Catania's Pizzeria & Cafe Inc.
Bella Pizza
My Place Family Pizza
Jerome's Pizza
Bedford Pizza
Rocco's Pizza Restaurant
The Famous Emilio's Pizza
Filomena's Pizza & Pasta
Carbo's Pizzeria
Screamer's Pizzeria
Russ Pizza
Italy Pizza
Saraghina
Bravi Ragazzi
Rocco Pizza III
Domino's Pizza
Little Caesars Pizza
Little Caesars Pizza
Papa John's Pizza
F&F Pizzeria
Brooklyn Pizza Market
Giardini Pizza
Via Roma Pizza Bar
Charles's Pizzeria
Luigi's Pizza
Royal Kings Pizza
Dominick's Pizza
Calabria
Basil Pizza & Wine Bar
Kingston Pizza
Pomodori
Calabria
Basil Pizza & Wine Bar
Kingston Pizza
Pomodori
Peppino's
Vesuvio Pizzeria & Restaurant
Rocco's Pizzeria
Lenny's Pizza
Pizza
Grotta Azzurra
Piccolo's
Jentana's Pizza
Amnon's Pizza
Pizza
Grotta Azzurra
Lenny's Pizza
La Bella Pizza Exp

In [160]:
pp_result

Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0         Linda's Pizzeria              40.823803              -73.909060   
1         Linda's Pizzeria              40.823803              -73.909060   
2         Linda's Pizzeria              40.823803              -73.909060   
3         Linda's Pizzeria              40.823803              -73.909060   
4         Linda's Pizzeria              40.823803              -73.909060   
5         Linda's Pizzeria              40.823803              -73.909060   
6         Linda's Pizzeria              40.823803              -73.909060   
7         Linda's Pizzeria              40.823803              -73.909060   
8         Linda's Pizzeria              40.823803              -73.909060   
9         Linda's Pizzeria              40.823803              -73.909060   
10        Linda's Pizzeria              40.823803              -73.909060   
11        Linda's Pizzeria              40.823803              -73.909060   
12        Linda's Pizzeria              40.823803              -73.909060   
13        Linda's Pizzeria              40.823803              -73.909060   
14        Linda's Pizzeria              40.823803              -73.909060   
15        Linda's Pizzeria              40.823803              -73.909060   
16        Linda's Pizzeria              40.823803              -73.909060   
17        Linda's Pizzeria              40.823803              -73.909060   
18        Linda's Pizzeria              40.823803              -73.909060   
19        Linda's Pizzeria              40.823803              -73.909060   
20        Linda's Pizzeria              40.823803              -73.909060   
21        Linda's Pizzeria              40.823803              -73.909060   
22        Linda's Pizzeria              40.823803              -73.909060   
23        Linda's Pizzeria              40.823803              -73.909060   
24        Linda's Pizzeria              40.823803              -73.909060   
25        Linda's Pizzeria              40.823803              -73.909060   
26        Linda's Pizzeria              40.823803              -73.909060   
27        Linda's Pizzeria              40.823803              -73.909060   
28        Linda's Pizzeria              40.823803              -73.909060   
29    Little Caesars Pizza              40.820466              -73.912912   
...                    ...                    ...                     ...   
6001          Square Pizza              40.541462              -74.177444   
6002          Square Pizza              40.541462              -74.177444   
6003          Square Pizza              40.541462              -74.177444   
6004          Square Pizza              40.541462              -74.177444   
6005          Square Pizza              40.541462              -74.177444   
6006          Square Pizza              40.541462              -74.177444   
6007         Trio pizzeria              40.541478              -74.178428   
6008         Trio pizzeria              40.541478              -74.178428   
6009         Trio pizzeria              40.541478              -74.178428   
6010         Trio pizzeria              40.541478              -74.178428   
6011         Trio pizzeria              40.541478              -74.178428   
6012         Trio pizzeria              40.541478              -74.178428   
6013         Trio pizzeria              40.541478              -74.178428   
6014         Trio pizzeria              40.541478              -74.178428   
6015         Trio pizzeria              40.541478              -74.178428   
6016         Trio pizzeria              40.541478              -74.178428   
6017         Trio pizzeria              40.541478              -74.178428   
6018         Trio pizzeria              40.541478              -74.178428   
6019      The Pizza Parlor              40.540517              -74.176718   
6020      The Pizza Parlor              40.540517              -74.176718   
6021      

In [211]:
pp_result2=pp_result[['id','Venue Latitude','Venue Longitude']].drop_duplicates(['id'])
pp_result2

id  Venue Latitude  Venue Longitude
0     5956be26123a195de6701c2b       40.823887       -73.912910
1     4fa534cee4b0fed4819dc7d4       40.823433       -73.910940
2     4f0f4fe14fc6e3adf353a950       40.819543       -73.910554
3     5591837d498ee4167d48bb73       40.822674       -73.911592
4     553980e0498e68099069f65c       40.822894       -73.909565
5     542c31e1498e76106c273492       40.819470       -73.910522
6     4d38c0949ca8236a1b12aee8       40.824605       -73.909819
7     4c6d654f65eda093577b4ed0       40.825183       -73.908625
8     4ca84cbc14c337040abcd23b       40.823888       -73.909905
9     4c7d00cf744d8cfacd1dbc37       40.824453       -73.909197
10    4e011a3a45dd1e4999dceb4e       40.823881       -73.909203
11    4bded394fe0e62b5b8fa0506       40.827092       -73.912007
12    4d111879a3d9721e4836d9fd       40.825447       -73.908951
13    4efdb8a129c201c6fb475af9       40.823106       -73.909023
14    4e4cf9c1bd413c4cc66dae95       40.823803       -73.909060
15    4dfccde4fa76c83b6e7de08d       40.822373       -73.909663
16    50bcf3a6e4b018e045e99570       40.824079       -73.906252
17    4ee659190e61681b98037e68       40.820384       -73.910185
18    517502d7498e616e6c4cb630       40.823266       -73.905368
19    4df7be37aeb7f7c3b5436f7a       40.820466       -73.912912
20    563ba7d9cd109b358ed35cd5       40.824657       -73.913605
21    4eb97c7c29c212fb6e1c9a7f       40.823567       -73.904266
22    4e6136ad81301b932281d077       40.824153       -73.913848
23    4fce5286e4b0921c9a424382       40.824196       -73.913848
24    515c85afe4b0b0da804faa30       40.821150       -73.912436
25    4e50fedad22daf51d28b3486       40.823989       -73.914124
26    4c69b4ec3bad2d7fcd94b0ee       40.820330       -73.906884
27    4e5066131495fc2ca40dec52       40.823701       -73.914211
28    5107db17e4b07385651f27bb       40.821285       -73.904972
29    4ef4f48277c8e91264361790       40.820028       -73.915886
...                        ...             ...              ...
5972  4def5c91c65bf3f03e93bed9       40.596831       -74.070316
5973  4c96c79482b56dcbd0bde4aa       40.598734       -74.062288
5974  4c96cc4d38dd8cfaf9ccde62       40.598143       -74.066233
5975  4ee8fe31754ab0f26061b313       40.598461       -74.065236
5976  4b4b6a9ef964a520c79a26e3       40.598126       -74.066423
5977  4dc53a75fa76d685cdda30ea       40.598814       -74.063710
5978  4f7c9f93e4b06b6ab9b51813       40.599184       -74.063899
5979  4c4251fed691c9b688388e0a       40.596786       -74.070130
5980  4ce736b7f3bda1439611b1e4       40.600412       -74.064524
5981  4c292b27ce3fc928a3596e88       40.595466       -74.064197
5982  4bd2dc9277b29c7401738f82       40.594412       -74.068013
5983  4c2e6d7ca0ced13a2fd80e6e       40.598398       -74.072217
5984  500d3465e4b0ba42c2e43052       40.598350       -74.072314
5985  4e08d6828877394591eea17c       40.598651       -74.072297
5986  4c092286a1b32d7f571897f0       40.598048       -74.072645
5988  4d6eae912427224ba1fed04d       40.601386       -74.071953
5989  4edfd9e7f9aba0b103003cb6       40.603417       -74.070110
5994  575aed29498e56fd382c1a58       40.604531       -74.069226
5995  53944df4498ea90b2e5749fd       40.542555       -74.177187
5996  4aa06637f964a5207c3f20e3       40.542079       -74.177325
5997  4d02d39da2685481dc2baebd       40.540418       -74.177196
5998  4e3c7b181838cb1b82e368e5       40.541286       -74.178489
5999  4b915d10f964a52062b633e3       40.540482       -74.178185
6000  4be57392bcef2d7f83c703e5       40.541611       -74.177102
6001  4bbd1899a8cf76b02169b1fd       40.540475       -74.176878
6002  4ce9b5ffbaa6a1cd1d59356c       40.541478       -74.178428
6003  4b5e1217f964a520fd7b29e3       40.540604       -74.176116
6004  580e160438fa564f926c7fa0       40.541654       -74.177226
6005  58eea452e72ef631e9b864c8       40.540517       -74.176718
6006  53ebf523498ee86ebe0de1e9       40.541292       -74.177928

[3249 rows x 3 columns]

In [167]:
pp_venues = pd.get_dummies(pp_result[['Venue Category']], prefix="", prefix_sep="")
pp_venues

Accessories Store  Afghan Restaurant  African Restaurant  \
0                     0                  0                   0   
1                     0                  0                   0   
2                     0                  0                   0   
3                     0                  0                   0   
4                     0                  0                   0   
5                     0                  0                   0   
6                     0                  0                   0   
7                     0                  0                   0   
8                     0                  0                   0   
9                     0                  0                   0   
10                    0                  0                   0   
11                    0                  0                   0   
12                    0                  0                   0   
13                    0                  0                   0   
14                    0                  0                   0   
15                    0                  0                   0   
16                    0                  0                   0   
17                    0                  0                   0   
18                    0                  0                   0   
19                    0                  0                   0   
20                    0                  0                   0   
21                    0                  0                   0   
22                    0                  0                   0   
23                    0                  0                   0   
24                    0                  0                   0   
25                    0                  0                   0   
26                    0                  0                   0   
27                    0                  0                   0   
28                    0                  0                   0   
29                    0                  0                   0   
...                 ...                ...                 ...   
6001                  0                  0                   0   
6002                  0                  0                   0   
6003                  0                  0                   0   
6004                  0                  0                   0   
6005                  0                  0                   0   
6006                  0                  0                   0   
6007                  0                  0                   0   
6008                  0                  0                   0   
6009                  0                  0                   0   
6010                  0                  0                   0   
6011                  0                  0                   0   
6012                  0                  0                   0   
6013                  0                  0                   0   
6014                  0                  0                   0   
6015                  0                  0                   0   
6016                  0                  0                   0   
6017                  0                  0                   0   
6018                  0                  0                   0   
6019                  0                  0                   0   
6020                  0                  0                   0   
6021                  0                  0                   0   
6022                  0                  0                   0   
6023                  0                  0                   0   
6024                  0                  0                   0   
6025                  0                  0                   0   
6026                  0                  0                   0   
6027                  0                  0                   0   
6028                  0                  0                   0   
6029                  0                  0    

In [168]:
pp_venues.insert(0,'PizzaPlace',pp_result['Neighborhood'])
pp_venues

PizzaPlace  Accessories Store  Afghan Restaurant  \
0         Linda's Pizzeria                  0                  0   
1         Linda's Pizzeria                  0                  0   
2         Linda's Pizzeria                  0                  0   
3         Linda's Pizzeria                  0                  0   
4         Linda's Pizzeria                  0                  0   
5         Linda's Pizzeria                  0                  0   
6         Linda's Pizzeria                  0                  0   
7         Linda's Pizzeria                  0                  0   
8         Linda's Pizzeria                  0                  0   
9         Linda's Pizzeria                  0                  0   
10        Linda's Pizzeria                  0                  0   
11        Linda's Pizzeria                  0                  0   
12        Linda's Pizzeria                  0                  0   
13        Linda's Pizzeria                  0                  0   
14        Linda's Pizzeria                  0                  0   
15        Linda's Pizzeria                  0                  0   
16        Linda's Pizzeria                  0                  0   
17        Linda's Pizzeria                  0                  0   
18        Linda's Pizzeria                  0                  0   
19        Linda's Pizzeria                  0                  0   
20        Linda's Pizzeria                  0                  0   
21        Linda's Pizzeria                  0                  0   
22        Linda's Pizzeria                  0                  0   
23        Linda's Pizzeria                  0                  0   
24        Linda's Pizzeria                  0                  0   
25        Linda's Pizzeria                  0                  0   
26        Linda's Pizzeria                  0                  0   
27        Linda's Pizzeria                  0                  0   
28        Linda's Pizzeria                  0                  0   
29    Little Caesars Pizza                  0                  0   
...                    ...                ...                ...   
6001          Square Pizza                  0                  0   
6002          Square Pizza                  0                  0   
6003          Square Pizza                  0                  0   
6004          Square Pizza                  0                  0   
6005          Square Pizza                  0                  0   
6006          Square Pizza                  0                  0   
6007         Trio pizzeria                  0                  0   
6008         Trio pizzeria                  0                  0   
6009         Trio pizzeria                  0                  0   
6010         Trio pizzeria                  0                  0   
6011         Trio pizzeria                  0                  0   
6012         Trio pizzeria                  0                  0   
6013         Trio pizzeria                  0                  0   
6014         Trio pizzeria                  0                  0   
6015         Trio pizzeria                  0                  0   
6016         Trio pizzeria                  0                  0   
6017         Trio pizzeria                  0                  0   
6018         Trio pizzeria                  0                  0   
6019      The Pizza Parlor                  0                  0   
6020      The Pizza Parlor                  0                  0   
6021      The Pizza Parlor                  0                  0   
6022      The Pizza Parlor                  0                  0   
6023      The Pizza Parlor                  0                  0   
6024      The Pizza Parlor                  0                  0   
6025      The Pizza Parlor                  0                  0   
6026      The Pizza Parlor                  0                  0   
6027      The Pizza Parlor                  0                  0   
6028

In [186]:
pp_grouped = pp_venues.groupby('PizzaPlace').sum().reset_index()
pp_grouped

PizzaPlace  Accessories Store  \
0                              Amnon's Pizza                  0   
1                Angelo's Pizza & Restaurant                  0   
2                     Basil Pizza & Wine Bar                  0   
3                              Bedford Pizza                  0   
4                                Bella Pizza                  0   
5                            Bellini's Pizza                  0   
6                         Best Italian Pizza                  0   
7                              Bravi Ragazzi                  0   
8                          Brooklyn Pizza II                  0   
9                      Brooklyn Pizza Market                  0   
10                         Brooklyn Pizzeria                  0   
11          CHUBBY BURGERS chicken and pizza                  0   
12                                  Calabria                  0   
13                          Carbo's Pizzeria                  0   
14            Catania's Pizzeria & Cafe Inc.                  0   
15                        Charles's Pizzeria                  0   
16                                  Ciccio's                  0   
17  D'Angelo's Italian Pizzeria & Restaurant                  0   
18                          Dominick's Pizza                  0   
19                            Domino's Pizza                  1   
20                          Donna Margherita                  0   
21                              F&F Pizzeria                  0   
22                              Family Pizza                  0   
23                  Filomena's Pizza & Pasta                  0   
24                      Fratellis Pizza Cafe                  0   
25                        Full Moon Pizzeria                  0   
26                            Giardini Pizza                  0   
27                           Gino's Pizzeria                  0   
28              Gino's Pizzeria & Restaurant                  0   
29                              Gloria Pizza                  0   
..                                       ...                ...   
68                                Pizza Park                  0   
69                                  Pomodori                  0   
70                                     Posto                  0   
71                             Pugsley Pizza                  0   
72                           Rocco Pizza III                  0   
73                  Rocco's Pizza Restaurant                  0   
74                          Rocco's Pizzeria                  0   
75                         Royal Kings Pizza                  0   
76                                Russ Pizza                  0   
77                       S & S Calabro Pizza                  0   
78                         Sal & Pauls Pizza                  0   
79                     Sam's Famous Pizzeria                  0   
80                         San Remo Pizzeria                  0   
81                                 Saraghina                  0   
82                       Screamer's Pizzeria                  0   
83               Sottocasa Pizzeria - Harlem                  0   
84                              Square Pizza                  0   
85                             Susie's Pizza                  0   
86                 The Famous Emilio's Pizza                  0   
87                          The Pizza Parlor                  0   
88              Tony's Pizzeria & Restaurant                  0   
89                             Trio pizzeria                  0   
90             Vesuvio Pizzeria & Restaurant                  0   
91                        Via Ponte Pizzeria                  0   
92                        Via Roma Pizza Bar                  0   
93                            Vidali's Pizza                  0   
94                     Vittorio's Pizza Plus                  0   
95                            Zero Otto Nove                  0   
96            

# Clustering
Cluster Pizza Places into 5 types according to the total numbers of different venues around.

In [187]:
# set number of clusters
kclusters2 = 5
pp_clustering2 = pp_grouped.drop(['PizzaPlace'], 1)
# run k-means clustering
kmeans2 = KMeans(n_clusters=kclusters2, random_state=0).fit(pp_clustering2)

# check cluster labels generated for each row in the dataframe
kmeans2.labels_[-10:-1] 

array([3, 3, 0, 3, 0, 3, 3, 1, 0])

In [188]:
pp_grouped.insert(1, 'ClusterLabels', kmeans2.labels_)
pp_grouped

PizzaPlace  ClusterLabels  \
0                              Amnon's Pizza              3   
1                Angelo's Pizza & Restaurant              3   
2                     Basil Pizza & Wine Bar              0   
3                              Bedford Pizza              3   
4                                Bella Pizza              1   
5                            Bellini's Pizza              1   
6                         Best Italian Pizza              3   
7                              Bravi Ragazzi              3   
8                          Brooklyn Pizza II              3   
9                      Brooklyn Pizza Market              0   
10                         Brooklyn Pizzeria              3   
11          CHUBBY BURGERS chicken and pizza              3   
12                                  Calabria              0   
13                          Carbo's Pizzeria              3   
14            Catania's Pizzeria & Cafe Inc.              1   
15                        Charles's Pizzeria              3   
16                                  Ciccio's              3   
17  D'Angelo's Italian Pizzeria & Restaurant              3   
18                          Dominick's Pizza              3   
19                            Domino's Pizza              2   
20                          Donna Margherita              0   
21                              F&F Pizzeria              0   
22                              Family Pizza              0   
23                  Filomena's Pizza & Pasta              3   
24                      Fratellis Pizza Cafe              3   
25                        Full Moon Pizzeria              1   
26                            Giardini Pizza              0   
27                           Gino's Pizzeria              0   
28              Gino's Pizzeria & Restaurant              3   
29                              Gloria Pizza              3   
..                                       ...            ...   
68                                Pizza Park              0   
69                                  Pomodori              3   
70                                     Posto              0   
71                             Pugsley Pizza              0   
72                           Rocco Pizza III              0   
73                  Rocco's Pizza Restaurant              3   
74                          Rocco's Pizzeria              3   
75                         Royal Kings Pizza              3   
76                                Russ Pizza              0   
77                       S & S Calabro Pizza              3   
78                         Sal & Pauls Pizza              3   
79                     Sam's Famous Pizzeria              0   
80                         San Remo Pizzeria              3   
81                                 Saraghina              3   
82                       Screamer's Pizzeria              0   
83               Sottocasa Pizzeria - Harlem              0   
84                              Square Pizza              3   
85                             Susie's Pizza              3   
86                 The Famous Emilio's Pizza              3   
87                          The Pizza Parlor              3   
88              Tony's Pizzeria & Restaurant              3   
89                             Trio pizzeria              3   
90             Vesuvio Pizzeria & Restaurant              0   
91                        Via Ponte Pizzeria              3   
92                        Via Roma Pizza Bar              0   
93                            Vidali's Pizza              3   
94                     Vittorio's Pizza Plus              3   
95                            Zero Otto Nove              1   
96                         ZuriLee Pizza Bar              0   
97                                babbalucci              0   

    Accessories Store  Afghan Restaurant  African Restaurant  \
0                   0                  0                   0   

In [216]:
pp_final=pp_grouped[['PizzaPlace','ClusterLabels']].merge(pp_dff,left_on='PizzaPlace',right_on='Name')
pp_final=pp_final.drop(['Name'],1)
pp_final=pp_final.merge(pp_result2,on=['id'])
pp_final=pp_final.sort_values(by=['Rating'],ascending=False)
pp_final

PizzaPlace  ClusterLabels  \
90                Sottocasa Pizzeria - Harlem              0   
88                                  Saraghina              3   
63                               Nick's Pizza              3   
98                             Zero Otto Nove              1   
79                                      Posto              0   
100                                babbalucci              0   
47                          La Villa Pizzeria              3   
73                                  Peppino's              3   
93              Vesuvio Pizzeria & Restaurant              0   
23                               F&F Pizzeria              0   
27                         Full Moon Pizzeria              1   
1                 Angelo's Pizza & Restaurant              3   
80                              Pugsley Pizza              0   
45                        La Pizza & La Pasta              0   
56                                 Lombardi’s              0   
99                          ZuriLee Pizza Bar              0   
51                   Lions & Tigers & Squares              0   
30               Gino's Pizzeria & Restaurant              3   
8                       Brooklyn Pizza Market              0   
35                                     Gruppo              0   
94                         Via Ponte Pizzeria              3   
22                           Donna Margherita              0   
76                                 Pizza Park              0   
86                      Sam's Famous Pizzeria              0   
89                        Screamer's Pizzeria              0   
4                             Bellini's Pizza              1   
82                           Rocco's Pizzeria              3   
59                          Milano's Pizzeria              0   
37                            Jentana's Pizza              3   
64                                 Oath Pizza              0   
..                                        ...            ...   
26                       Fratellis Pizza Cafe              3   
25                   Filomena's Pizza & Pasta              3   
62                               Napoli Pizza              3   
15                                   Ciccio's              3   
20                             Domino's Pizza              2   
96                             Vidali's Pizza              3   
97                      Vittorio's Pizza Plus              3   
69                          Papa John's Pizza              2   
17                           Dominick's Pizza              3   
71                          Papa John's Pizza              2   
70                          Papa John's Pizza              2   
19                             Domino's Pizza              2   
21                             Domino's Pizza              2   
16   D'Angelo's Italian Pizzeria & Restaurant              3   
18                             Domino's Pizza              2   
50                                Lil Caesars              3   
40                             Kingston Pizza              3   
41                             Kingston Pizza              3   
72                          Papa John's Pizza              2   
67                          Papa John's Pizza              2   
53                       Little Caesars Pizza              0   
91                  The Famous Emilio's Pizza              3   
66                                  P&M Pizza              0   
68                          Papa John's Pizza              2   
58                               Mike's Pizza              0   
9                           Brooklyn Pizzeria              3   
61                       N&D pizza restaurant              3   
78                                   Pomodori              3   
77                                   Pomodori              3   
52                       Little Caesars Pizza              0   

                           id  Rating  Price  Likes  Photos  Tips 

Mark out these pizza places on a map. Differentiate them with different colors according to their cluster labels.

In [362]:
import folium
# create map
map_final = folium.Map(location=[40.7128, -74.0060], zoom_start=10)

# set color scheme for the clusters
x = np.arange(kclusters2)
ys = [i + x + (i*x)**2 for i in range(kclusters2)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(pp_final['Venue Latitude'], pp_final['Venue Longitude'],  pp_final['PizzaPlace'], pp_final['ClusterLabels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-2],
        fill=True,
        fill_color=rainbow[cluster-2],
        fill_opacity=0.7).add_to(map_final)
       
map_final

Then compare ratings among these 5 groups.

In [221]:
pp_exp1=pp_final.groupby(['ClusterLabels'])['Rating'].median()
pp_exp1

ClusterLabels
0    8.0
1    8.4
2    6.4
3    7.4
4    7.8
Name: Rating, dtype: float64

In [227]:
import statistics 
pp_exp2=pp_final.groupby(['ClusterLabels'])['Price'].apply(statistics.mode)
pp_exp2

ClusterLabels
0    1
1    1
2    1
3    1
4    1
Name: Price, dtype: int64

In [228]:
 from scipy import stats

In [240]:
 stats.f_oneway(pp_final['ClusterLabels'],pp_final['Rating'])

F_onewayResult(statistic=1043.705817516981, pvalue=2.6288992377794986e-81)

In [244]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
from statsmodels.stats.multicomp import MultiComparison
mc=MultiComparison(pp_final['Rating'],pp_final['ClusterLabels'])

print(mc.tukeyhsd())
print(mc.groupsunique)

Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower   upper  reject
----------------------------------------------------
     0      1   0.3771    0.9 -0.8571  1.6114  False
     0      2  -1.4329  0.001 -2.2714 -0.5944   True
     0      3  -0.5429 0.0402 -1.0699 -0.0158   True
     0      4  -0.1371    0.9 -1.1054  0.8311  False
     1      2    -1.81  0.004 -3.1935 -0.4265   True
     1      3    -0.92 0.2302 -2.1401  0.3001  False
     1      4  -0.5143  0.851   -1.98  0.9514  False
     2      3     0.89 0.0258  0.0725  1.7075   True
     2      4   1.2957 0.0194  0.1433  2.4481   True
     3      4   0.4057 0.7322 -0.5444  1.3558  False
----------------------------------------------------
[0 1 2 3 4]


The performances of these 5 types pizza places are significantly different. Places with cluster label 0 and 1 perform best while those with label 2 perform worst. Hence, different neigbhorhood might affect the impression of the customers on this restaurant. Restaurant owners may search for a similar location to start their business. 
So what is special in these locations around pizza places?

In [246]:
pp_cb=pp_df[['Neighborhood','id']].merge(pp_final,on='id')
pp_cb

Neighborhood                        id  \
0            Bronx CB 1  4df7be37aeb7f7c3b5436f7a   
1            Bronx CB 2  4ca3e96c5720b1f78d0936ef   
2            Bronx CB 4  4afac66af964a520a11822e3   
3            Bronx CB 4  4afac66af964a520a11822e3   
4            Bronx CB 7  4afac66af964a520a11822e3   
5            Bronx CB 7  4afac66af964a520a11822e3   
6            Bronx CB 5  4bf329ef2d62952165ec5f58   
7            Bronx CB 5  4cb0f72d75ebb60cba60c9ad   
8            Bronx CB 6  4aabd3e6f964a5204a5a20e3   
9            Bronx CB 6  4aee1220f964a520ced121e3   
10           Bronx CB 6  4b140a90f964a520489c23e3   
11           Bronx CB 6  4b2a6ba8f964a520d9a824e3   
12           Bronx CB 6  4d03ef588620224bff0ba640   
13           Bronx CB 7  4bd71c916798ef3b563a668d   
14           Bronx CB 9  5169feaae4b04280a40bcd5a   
15          Bronx CB 10  4c27a574c11dc9b634f32a24   
16        Brooklyn CB 1  5761f36b498e2f6dca191e23   
17        Brooklyn CB 1  4ab947baf964a520c17e20e3   
18        Brooklyn CB 1  4a142410f964a52007781fe3   
19        Brooklyn CB 3  4a593de0f964a52015b91fe3   
20        Brooklyn CB 3  58bad1af45c3ed4ca99f97f9   
21        Brooklyn CB 3  4ecc3ea22c5b26aab86608f8   
22        Brooklyn CB 3  4b88126af964a52096dd31e3   
23        Brooklyn CB 3  4e3f52476365a251c704391c   
24        Brooklyn CB 5  4eb5e667b8030fb583171e69   
25        Brooklyn CB 5  4c88221997828cfab0b399aa   
26        Brooklyn CB 6  5d56bcea6ac2400008de14d9   
27        Brooklyn CB 6  596ab0631de7657c78f4251b   
28        Brooklyn CB 6  49d7e8c0f964a520795d1fe3   
29        Brooklyn CB 6  58016254d67c1565bc122795   
..                  ...                       ...   
87       Manhattan CB 4  5aea805d2b9844002c1bfe6e   
88       Manhattan CB 6  43c4bb46f964a5205a2d1fe3   
89       Manhattan CB 8  5893583c1470b050ba899197   
90       Manhattan CB 8  4a72520af964a520e2da1fe3   
91       Manhattan CB 8  5b33c56f06fb600032c88e26   
92       Manhattan CB 8  4c0fde4b81e976b0623c0eeb   
93       Manhattan CB 9  4d5028cedcce224bb140d51b   
94       Manhattan CB 9  5812ea4538fa932fe0cc3951   
95       Manhattan CB 9  505380aee4b07ab78c6545fe   
96       Manhattan CB 9  4df94db2a809fe7742217cf6   
97      Manhattan CB 10  55a4456a498e29417f5c71a2   
98      Manhattan CB 10  56e86a57cd1017cb53f3e8f9   
99      Manhattan CB 10  4ec312c6be7b4ee6107f53be   
100     Manhattan CB 11  4bffe7dd92a6c928d44443e2   
101     Manhattan CB 11  4ac54941f964a5205cb120e3   
102     Manhattan CB 11  4bae85aff964a5206bbc3be3   
103     Manhattan CB 11  4c0ab529a1b32d7fd8d599f0   
104     Manhattan CB 12  4ab97430f964a520727f20e3   
105         Queens CB 1  4bf2ad7782002d7fd5cfd0bb   
106         Queens CB 1  50ece8c6e4b0ffbaa8a65f57   
107         Queens CB 4  4bcb3f9ccc8cd13aa0f2becf   
108         Queens CB 4  4b7c363df964a520ce842fe3   
109         Queens CB 4  4d3bd7f2fb4c5481ca6ccf14   
110         Queens CB 6  4f6cb7a0e4b0a6b770c9b395   
111         Queens CB 6  45ac11b0f964a5205b411fe3   
112         Queens CB 9  4baebed0f964a5204ad43be3   
113         Queens CB 9  514cb2f3e4b07ece6ded65b5   
114        Queens CB 10  4b9c43bdf964a5202e5a36e3   
115        Queens CB 11  515a2513e4b0d8e05fce8144   
116  Staten Island CB 2  4c0806a4009a0f473473e5bf   

                                   PizzaPlace  ClusterLabels  Rating  Price  \
0                        Little Caesars Pizza              0     5.7      1   
1                        Fratellis Pizza Cafe              3     6.8      1   
2                              Jerome's Pizza              3     7.5      1   
3                              Jerome's Pizza              3     7.5      1   
4                              Jerome's Pizza              3     7.5      1   
5                              Jerome's Pizza              3     7.5      1   
6                          Best Italian Pizza              3     7.5      1   
7                              Domino's Pizza              2     6.3      1   
8            

In [310]:
pp_cb_r2=pp_cb.groupby(['Neighborhood'])['Rating'].mean()
pp_cb_r2=pd.DataFrame(pp_cb_r2)

In [317]:
pp_cb_f=df2.merge(pp_cb_r2,right_on='Neighborhood',left_on='Community Board')
pp_cb_f

Community Board Area/km2 Pop.Census Pop/km2  \
0           Bronx CB 1     7.17     91,497  12,761   
1           Bronx CB 2     5.54     52,246   9,792   
2           Bronx CB 4     5.28    146,441  27,735   
3           Bronx CB 5     3.55    128,200  36,145   
4           Bronx CB 6     4.01     83,268  20,765   
5           Bronx CB 7     4.84    139,286  28,778   
6           Bronx CB 9    12.41    172,298  13,884   
7          Bronx CB 10    16.76    120,392   7,183   
8        Brooklyn CB 1    12.82    160,338  12,507   
9        Brooklyn CB 3     7.67    143,867  18,757   
10       Brooklyn CB 5    14.61    173,198  11,855   
11       Brooklyn CB 6     9.01    104,054  11,549   
12       Brooklyn CB 7    10.96    120,063  10,955   
13       Brooklyn CB 8     4.25     96,076  22,606   
14       Brooklyn CB 9     4.07    104,014  25,556   
15      Brooklyn CB 10    10.57    122,542  11,593   
16      Brooklyn CB 11    10.18    172,129  16,909   
17      Brooklyn CB 12     9.32    185,046  19,855   
18      Brooklyn CB 13     8.88    106,120  11,950   
19      Brooklyn CB 14     7.61    168,806  22,182   
20      Brooklyn CB 15    12.82    160,319  12,505   
21      Brooklyn CB 16     4.97     85,343  17,172   
22      Brooklyn CB 17     8.73    165,753  18,987   
23      Brooklyn CB 18    24.68    194,653   7,887   
24      Manhattan CB 1     4.45     34,420   7,735   
25      Manhattan CB 3     4.56    164,407  36,054   
26      Manhattan CB 4     5.41     87,479  16,170   
27      Manhattan CB 6     3.55    136,152  38,353   
28      Manhattan CB 8     5.13    217,063  42,312   
29      Manhattan CB 9     3.91    111,724  28,574   
30     Manhattan CB 10     3.63    107,109  29,507   
31     Manhattan CB 11     5.75    117,743  20,477   
32     Manhattan CB 12     7.64    208,414  27,279   
33         Queens CB 1    14.97    211,220  14,110   
34         Queens CB 4     6.16    167,005  27,111   
35         Queens CB 6     7.59    115,967  15,279   
36         Queens CB 9    10.08    141,608  14,048   
37        Queens CB 10    16.19    127,274   7,861   
38        Queens CB 11    24.99    116,404   4,658   
39  Staten Island CB 2    54.62    127,071   2,326   

                                        Neighborhoods   Latitude  Longitude  \
0                    Melrose, Mott Haven, Port Morris  40.824545 -73.910414   
1                               Hunts Point, Longwood  40.812601 -73.884025   
2                               Concourse, Highbridge  40.874217 -73.890410   
3   Fordham, Morris Heights, Mount Hope, Universit...  40.859267 -73.898469   
4         Bathgate, Belmont, East Tremont, West Farms  40.855278 -73.886389   
5           Bedford Park, Norwood, University Heights  40.870100 -73.885691   
6   Bronx River, Bruckner, Castle Hill, Clason Poi...  40.850485 -73.840404   
7   City Island, Co-op City, Locust Point, Pelham ...  40.847456 -73.786466   
8                            Greenpoint, Williamsburg  40.723713 -73.950971   
9   Bedford-Stuyvesant, Ocean Hill, Stuyvesant Hei...  40.683436 -73.941249   
10  City Line, Cypress Hills, East New York, Highl...  40.676225 -73.886962   
11  Carroll Gardens, Cobble Hill, Gowanus, Park Sl...  40.678420 -73.994802   
12    Greenwood Heights, Sunset Park, Windsor Terrace  40.644337 -74.007532   
13        Crown Heights, Prospect Heights, Weeksville  40.667471 -73.943566   
14  Crown Heights, Prospect Lefferts Gardens, Wingate  40.667471 -73.943566   
15            Bay Ridge, Dyker Heights, Fort Hamilton  40.630188 -74.027190   
16       Bath Beach, Bensonhurst, Gravesend, Mapleton  40.601850 -74.000501   
17   Borough Park, Kensington, Midwood, Ocean Parkway  40.633993 -73.996806   
18  Bensonhurst, Brighton Beach, Coney Island, Gra...  40.604977 -73.993406   
19       Flatbush, Kensington, Midwood, Ocean Parkway  40.652048 -73.959027   
20  East Gravesend, Gerritsen Beach, Homecrest, Ki...  40.596134 -73.973943   
21                            Brownsville, Ocean Hill 

In [318]:
pp_cb_f['Pop/km2']=pp_cb_f['Pop/km2'].apply(lambda s: s.replace(',',''))
pp_cb_f['Pop.Census']=pp_cb_f['Pop.Census'].apply(lambda s: s.replace(',',''))
pp_cb_f['Area/km2']=pp_cb_f['Area/km2'].apply(lambda s: s.replace(',',''))
pd.to_numeric(pp_cb_f['Pop/km2'])
pd.to_numeric(pp_cb_f['Area/km2'])
pd.to_numeric(pp_cb_f['Pop.Census'])
pp_cb_f

Community Board Area/km2 Pop.Census Pop/km2  \
0           Bronx CB 1     7.17      91497   12761   
1           Bronx CB 2     5.54      52246    9792   
2           Bronx CB 4     5.28     146441   27735   
3           Bronx CB 5     3.55     128200   36145   
4           Bronx CB 6     4.01      83268   20765   
5           Bronx CB 7     4.84     139286   28778   
6           Bronx CB 9    12.41     172298   13884   
7          Bronx CB 10    16.76     120392    7183   
8        Brooklyn CB 1    12.82     160338   12507   
9        Brooklyn CB 3     7.67     143867   18757   
10       Brooklyn CB 5    14.61     173198   11855   
11       Brooklyn CB 6     9.01     104054   11549   
12       Brooklyn CB 7    10.96     120063   10955   
13       Brooklyn CB 8     4.25      96076   22606   
14       Brooklyn CB 9     4.07     104014   25556   
15      Brooklyn CB 10    10.57     122542   11593   
16      Brooklyn CB 11    10.18     172129   16909   
17      Brooklyn CB 12     9.32     185046   19855   
18      Brooklyn CB 13     8.88     106120   11950   
19      Brooklyn CB 14     7.61     168806   22182   
20      Brooklyn CB 15    12.82     160319   12505   
21      Brooklyn CB 16     4.97      85343   17172   
22      Brooklyn CB 17     8.73     165753   18987   
23      Brooklyn CB 18    24.68     194653    7887   
24      Manhattan CB 1     4.45      34420    7735   
25      Manhattan CB 3     4.56     164407   36054   
26      Manhattan CB 4     5.41      87479   16170   
27      Manhattan CB 6     3.55     136152   38353   
28      Manhattan CB 8     5.13     217063   42312   
29      Manhattan CB 9     3.91     111724   28574   
30     Manhattan CB 10     3.63     107109   29507   
31     Manhattan CB 11     5.75     117743   20477   
32     Manhattan CB 12     7.64     208414   27279   
33         Queens CB 1    14.97     211220   14110   
34         Queens CB 4     6.16     167005   27111   
35         Queens CB 6     7.59     115967   15279   
36         Queens CB 9    10.08     141608   14048   
37        Queens CB 10    16.19     127274    7861   
38        Queens CB 11    24.99     116404    4658   
39  Staten Island CB 2    54.62     127071    2326   

                                        Neighborhoods   Latitude  Longitude  \
0                    Melrose, Mott Haven, Port Morris  40.824545 -73.910414   
1                               Hunts Point, Longwood  40.812601 -73.884025   
2                               Concourse, Highbridge  40.874217 -73.890410   
3   Fordham, Morris Heights, Mount Hope, Universit...  40.859267 -73.898469   
4         Bathgate, Belmont, East Tremont, West Farms  40.855278 -73.886389   
5           Bedford Park, Norwood, University Heights  40.870100 -73.885691   
6   Bronx River, Bruckner, Castle Hill, Clason Poi...  40.850485 -73.840404   
7   City Island, Co-op City, Locust Point, Pelham ...  40.847456 -73.786466   
8                            Greenpoint, Williamsburg  40.723713 -73.950971   
9   Bedford-Stuyvesant, Ocean Hill, Stuyvesant Hei...  40.683436 -73.941249   
10  City Line, Cypress Hills, East New York, Highl...  40.676225 -73.886962   
11  Carroll Gardens, Cobble Hill, Gowanus, Park Sl...  40.678420 -73.994802   
12    Greenwood Heights, Sunset Park, Windsor Terrace  40.644337 -74.007532   
13        Crown Heights, Prospect Heights, Weeksville  40.667471 -73.943566   
14  Crown Heights, Prospect Lefferts Gardens, Wingate  40.667471 -73.943566   
15            Bay Ridge, Dyker Heights, Fort Hamilton  40.630188 -74.027190   
16       Bath Beach, Bensonhurst, Gravesend, Mapleton  40.601850 -74.000501   
17   Borough Park, Kensington, Midwood, Ocean Parkway  40.633993 -73.996806   
18  Bensonhurst, Brighton Beach, Coney Island, Gra...  40.604977 -73.993406   
19       Flatbush, Kensington, Midwood, Ocean Parkway  40.652048 -73.959027   
20  East Gravesend, Gerritsen Beach, Homecrest, Ki...  40.596134 -73.973943   
21                            Brownsville, Ocean Hill 

In [330]:
pp_high=pp_grouped[pp_grouped['ClusterLabels']<=1]
pp_high

PizzaPlace  ClusterLabels  Accessories Store  \
2                    Basil Pizza & Wine Bar              0                  0   
4                               Bella Pizza              1                  0   
5                           Bellini's Pizza              1                  0   
9                     Brooklyn Pizza Market              0                  0   
12                                 Calabria              0                  0   
14           Catania's Pizzeria & Cafe Inc.              1                  0   
20                         Donna Margherita              0                  0   
21                             F&F Pizzeria              0                  0   
22                             Family Pizza              0                  0   
25                       Full Moon Pizzeria              1                  0   
26                           Giardini Pizza              0                  0   
27                          Gino's Pizzeria              0                  0   
32                                   Gruppo              0                  0   
33                              Italy Pizza              0                  0   
40  La Crosta Restaurant & Gourmet Pizzeria              0                  0   
42                      La Pizza & La Pasta              0                  0   
48                 Lions & Tigers & Squares              0                  0   
49                     Little Caesars Pizza              0                  0   
51                               Lombardi’s              0                  0   
53                           Margherita NYC              0                  0   
54                             Mike's Pizza              0                  0   
55                        Milano's Pizzeria              0                  0   
60                               Oath Pizza              0                  0   
62                                P&M Pizza              0                  0   
68                               Pizza Park              0                  0   
70                                    Posto              0                  0   
71                            Pugsley Pizza              0                  0   
72                          Rocco Pizza III              0                  0   
76                               Russ Pizza              0                  0   
79                    Sam's Famous Pizzeria              0                  0   
82                      Screamer's Pizzeria              0                  0   
83              Sottocasa Pizzeria - Harlem              0                  0   
90            Vesuvio Pizzeria & Restaurant              0                  0   
92                       Via Roma Pizza Bar              0                  0   
95                           Zero Otto Nove              1                  0   
96                        ZuriLee Pizza Bar              0                  0   
97                               babbalucci              0                  0   

    Afghan Restaurant  African Restaurant  American Restaurant  Amphitheater  \
2                   0                   0                    0             0   
4                   0                   0                    2             0   
5                   0                   0                    1             0   
9                   0                   0                    0             0   
12                  0                   0                    0             0   
14                  0                   0                    0             0   
20                  1                   0                    0             0   
21                  0                   0                    1             0   
22                  0                   0                    1             0   
25                  0                   0                    1             0   
26                  0                   0                    1            

In [339]:
pp_high=pp_high.ix[:,~(pp_high==0).all()]

C:\Users\Gary\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#ix-indexer-is-deprecated
  """Entry point for launching an IPython kernel.


In [340]:
del pp_high['PizzaPlace']

In [351]:
pp_unknown=pp_high.groupby('ClusterLabels').mean()
pp_unknown=numpy.transpose(pp_unknown)
pp_unknown

ClusterLabels                        0    1
Afghan Restaurant              0.12500  0.0
African Restaurant             0.25000  0.0
American Restaurant            0.90625  1.0
Amphitheater                   0.06250  0.0
Animal Shelter                 0.03125  0.0
Antique Shop                   0.12500  0.0
Arepa Restaurant               0.03125  0.0
Argentinian Restaurant         0.09375  0.0
Art Gallery                    0.37500  0.0
Art Museum                     0.12500  0.0
Arts & Crafts Store            0.37500  0.0
Asian Restaurant               0.18750  0.0
Athletics & Sports             0.06250  0.2
Auditorium                     0.03125  0.0
BBQ Joint                      0.09375  0.2
Baby Store                     0.06250  0.0
Bagel Shop                     1.06250  0.0
Bakery                         1.81250  5.2
Bank                           0.87500  1.0
Bar                            1.78125  1.4
Baseball Field                 0.03125  0.0
Baseball Stadium               0.03125  0.0
Basketball Court               0.03125  0.0
Beer Bar                       0.34375  0.0
Beer Garden                    0.21875  0.0
Beer Store                     0.06250  0.0
Big Box Store                  0.03125  0.0
Bike Shop                      0.03125  0.0
Bistro                         0.06250  0.0
Bookstore                      0.46875  0.0
...                                ...  ...
Supermarket                    0.53125  0.2
Supplement Shop                0.21875  0.2
Sushi Restaurant               1.40625  0.0
Swiss Restaurant               0.03125  0.0
Taco Place                     0.50000  0.0
Taiwanese Restaurant           0.06250  0.0
Tapas Restaurant               0.25000  0.0
Tattoo Parlor                  0.03125  0.4
Tea Room                       0.25000  0.0
Tech Startup                   0.06250  0.0
Tennis Court                   0.09375  0.0
Thai Restaurant                0.78125  0.0
Theater                        0.46875  0.0
Thrift / Vintage Store         0.59375  0.2
Tiki Bar                       0.18750  0.0
Toy / Game Store               0.09375  0.0
Track Stadium                  0.03125  0.0
Tree                           0.03125  0.0
Turkish Restaurant             0.18750  0.0
Vegetarian / Vegan Restaurant  0.62500  0.0
Video Game Store               0.12500  0.2
Video Store                    0.06250  0.0
Vietnamese Restaurant          0.21875  0.0
Waterfront                     0.03125  0.0
Whisky Bar                     0.03125  0.0
Wine Bar                       0.93750  0.0
Wine Shop                      1.31250  0.0
Wings Joint                    0.15625  0.4
Women's Store                  0.40625  0.2
Yoga Studio                    0.87500  0.0

[267 rows x 2 columns]

In [361]:
pp_unknown.sort_values(by=0 & 1, ascending=False).head(10)

ClusterLabels               0     1
Pizza Place           4.81250   5.8
Coffee Shop           3.53125   0.2
Deli / Bodega         2.59375   5.2
Italian Restaurant    2.37500  16.2
Café                  2.34375   1.0
Bakery                1.81250   5.2
Bar                   1.78125   1.4
Mexican Restaurant    1.59375   1.6
Caribbean Restaurant  1.46875   0.0
Sushi Restaurant      1.40625   0.0

Pizza place as a classic light meal, noramlly is opened at a location where different light meals gather, like Cafe, bakery or bar. It explaines why places with higher ratings gather around Manhattan. Does the pizza there truely taste better than other places'? Short time spent on tasting could be a reason that render the quality of ratings. Another interesting thing is that Italian restaurant numbers is apparently more than other types of restaurant, expecially for pizza places with label 1. These places are in Bronx which is happen to be an Italian area. Pizza as a representitive of Italian food seems meet the correct market there. 

# Conclusion
Therefore, for those types of pizza place where providing quick services and moderate flavor with moderate price, they should consider locations in busy area and close to other light meals restaurant. For those aiming to provide delicious pizza to most picky gourmet, they should open their pizza place at places close to their target customers or customers with certain background.